onnxscript/rewriter/pattern.py

```diff
@@ -328,13 +330,15 @@ def __init__(self) -> None:
         self.outputs: list[ir.Value] = []
         # For a failed match, _reason is a string that describes the reason for the failure.
         self._reason: str = ""
+        self._failure_node: ir.Node | None = None
```

Comment:
I would add a comment on what this field stores

Reply:
done